cmd/juju/application/storage.go

```diff
@@ -0,0 +1,267 @@
+		"yaml":    cmd.FormatYaml,
+	})
+}
+
+// Init implements part of the cmd.Command interface.
+func (c *storageConfigCommand) Init(args []string) error {
+	nArgs := len(args)
+	if nArgs == 0 {
+		return errors.Errorf("no application specified")
+	}
+
+	if !names.IsValidApplication(args[0]) {
+		return errors.Errorf("invalid application name %q", args[0])
+	}
+	c.applicationName = args[0]
+	return c.configBase.Init(args[1:])
+}
+
+// getAPI returns the API. This allows passing in a test configCommandAPI
+// implementation.
+func (c *storageConfigCommand) getAPI() (StorageConstraintsAPI, error) {
+	if c.api != nil {
+		return c.api, nil
+	}
+	api, err := c.NewAPIRoot()
+	if err != nil {
+		return nil, errors.Annotate(err, "opening API connection")
+	}
+	client := application.NewClient(api)
+	return client, nil
+}
+
+// Run implements the meaty part of the cmd.Command interface.
+func (c *storageConfigCommand) Run(ctx *cmd.Context) error {
+	client, err := c.getAPI()
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	for _, action := range c.configBase.Actions {
+		var err error
+		switch action {
+		case config.GetOne:
+			err = c.getConfig(client, ctx)
+		case config.SetArgs:
+			err = c.setConfig(client, c.configBase.ValsToSet)
+		default:
+			err = c.getAllConfig(client, ctx)
+		}
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+// StorageConstraintsAPI defines the API methods that the application storage command uses.
+type StorageConstraintsAPI interface {
+	Close() error
+	GetApplicationStorage(applicationName string) (application.ApplicationStorageInfo, error)
+	UpdateApplicationStorage(applicationStorageUpdateParams application.ApplicationStorageUpdate) error
+}
+
+// setConfig sets the provided key/value pairs on the application.
+func (c *storageConfigCommand) setConfig(client StorageConstraintsAPI, attrs config.Attrs) error {
+	sc := make(map[string]storage.Constraints)
+	for k, v := range attrs {
+		// This should give us a string of the form "100G,rootfs,1"
+		constraintsStr := fmt.Sprint(v)
+		parsedCons, err := storage.ParseConstraints(constraintsStr)
+		if err != nil {
+			return errors.Annotatef(err, "parsing storage constraints for %q", k)
+		}
+		sc[k] = parsedCons
+	}
+
+	updateParams := application.ApplicationStorageUpdate{
+		ApplicationTag:     names.NewApplicationTag(c.applicationName),
+		StorageConstraints: sc,
+	}
+
+	return block.ProcessBlockedError(client.UpdateApplicationStorage(updateParams), block.BlockChange)
+}
+
+// getConfig writes the value of a single application config key to the cmd.Context.
+func (c *storageConfigCommand) getConfig(client StorageConstraintsAPI, ctx *cmd.Context) error {
+	applicationStorageInfo, err := client.GetApplicationStorage(c.applicationName)
+	if err != nil {
+		return err
+	}
+
+	if len(c.configBase.KeysToGet) == 0 {
+		return errors.New("no storage key specified")
+	}
+
+	storeKey := c.configBase.KeysToGet[0]
```

Comment:
What happens when they ask for multiple storage keys ?

Reply:
It would be taking the getAllConfig path